### PR TITLE
FISH-8062 Third Party License Generation

### PR DESCRIPTION
**Pull request by kalinchan**

### CODE REVIEW

File: nucleus/osgi-platforms/felix/pom.xml [hunk 1]
```diff
@@ -82,6 +82,12 @@
     <build>
         <plugins>
             <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <version>2.3.0</version>
```
Comment (Pandrex247) [inline nucleus/osgi-platforms/felix/pom.xml:87]: Version not required, it's defined in `pluginManagement`

File: nucleus/pom.xml [hunk 2]
```diff
@@ -425,7 +425,11 @@
                     </execution>
                 </executions>
             </plugin>
-
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <version>2.3.0</version>
```
Comment (Pandrex247) [inline nucleus/pom.xml:431]: Version not required, it's defined in `pluginManagement`

File: pom.xml [hunk 3]
```diff
@@ -380,7 +380,27 @@
                         </testExcludes>
                     </configuration>
                 </plugin>
-
+                <plugin>
+                    <groupId>org.codehaus.mojo</groupId>
+                    <artifactId>license-maven-plugin</artifactId>
+                    <version>2.3.0</version>
```
Comment (Pandrex247) [inline pom.xml:386]: I'd prefer if this was defined as a property.

File: api/pom.xml [hunk 4]
```diff
@@ -69,6 +69,11 @@
                 </configuration>
                 <inherited>false</inherited>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <version>2.3.0</version>
```
Comment (Pandrex247) [inline api/pom.xml:75]: Version not required, it's defined in `pluginManagement`

File: appserver/packager/legal/src/main/resources/glassfish/legal/3RD-PARTY-LICENSE.txt
Comment (Pandrex247) [inline appserver/packager/legal/src/main/resources/glassfish/legal/3RD-PARTY-LICENSE.txt:1]: This doesn't seem to be autogenerated?
If I change a dependency version and rebuild the server the file still shows the old version

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: appserver/packager/legal/src/main/resources/glassfish/legal/3RD-PARTY-LICENSE.txt
Comment (Pandrex247) [reply appserver/packager/legal/src/main/resources/glassfish/legal/3RD-PARTY-LICENSE.txt:1]: Is there a reason we're hiding it behind a profile?

Comment (kalinchan) [reply appserver/packager/legal/src/main/resources/glassfish/legal/3RD-PARTY-LICENSE.txt:1]: Removed from profile

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
Can we add the 3RD-PARTY-LICENSE file to gitignore?
Does it need to be there at all?

After changing a version property (e.g. jackson 2.15.0 to 2.16.1) and building the server it shows the file as having changed in git

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**